### PR TITLE
BASIC-UI

### PR DESCRIPTION
**Pull request by Nowembabazi**
##In this milestone I achieved the following;
-Perfect workflow.
-Created an app.rb file.
-Followed the best practises of ruby.

### CODE REVIEW

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @Nowembabazi , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
- [ ] While creating person(student or teacher ), getting some argument error in initialisation.Plz correct it.
Screenshot from 2023-01-19 00-49-21

Screenshot from 2023-01-19 00-49-52


- NOTE:- I can't able to check the rental part as `person` is not created, plz check the rental part from your end before submitting it for the next review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: create_teacher.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  print 'Name: '
+  name = gets.chomp
+
+  print 'Specialization: '
+  specialization = gets.chomp
+
+  @persons.push(Teacher.new(specialization, age, name))
+  puts 'Teacher created successfully!'
+  puts ''
+end
```
Comment (zieeco) [inline create_teacher.rb:18]: - Great job addressing the issue pointed out by the previous reviewer, however, I can't create a teacher successfully. Please, kindly fix this issue of the argument below:


Screenshot from 2023-01-19 08-45-56

File: create_student.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    @persons.push(Student.new(true, age, name))
+    puts 'Student created successfully'
+  when 'n', 'N'
+    @persons.push(Student.new(false, age, name))
+    puts 'Student created successfully'
+  else
+    puts 'Invalid permission (Enter Y/N)'
+    library_menu
+  end
+end
```
Comment (zieeco) [inline create_student.rb:26]: - For the student, I don't seem to be able to create a student also. Please, fix the argument error it throws from the terminal below:


Screenshot from 2023-01-19 08-56-17

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Nowembabazi ,


Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the changes, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice app. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
-
     @person = person
-    person.rentals << self
-
     @book = book
-    book.rentals << self
+
+    @person.add_rental(self)
+    @book.add_rental(self)
   end
```
Comment (TedLivist) [inline rental.rb:11]: - [ ] When I try to create a rental I get this error message (see image). You can fix this by changing this `initialize` method to this as it was:

  ```
  def initialize(date, person, book)
    @date = date
    @person = person
    @book = book

    person.rentals << self
    book.rentals << self
  end
  ```

  image

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Nowembabazi,

✔️ Suggestion of option when listing books and people

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Nowembabazi,

✔️ Implementation of rental creation

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**